js/modules/k6/metrics/metrics.go

```diff
@@ -96,24 +103,59 @@ func (m Metric) GetName() string {
 	return m.metric.Name
 }
 
-type Metrics struct{}
+type (
+	RootMetricsModule struct{}
+	MetricsModule     struct {
+		getContext func() context.Context
+	}
+)
+
+func (m *MetricsModule) WithContext(getContext func() context.Context) {
+	m.getContext = getContext
+}
+
+func (*RootMetricsModule) NewModuleInstancePerVU() interface{} {
+	return &MetricsModule{}
```

Comment:
This is somewhat problematic as this :
1. exports withContext to the js which lets you do:
```
withContext(null)
```
bricking the module :facepalm: , this likely will only get worse with more `WIth` methods as we add
2. doesn't have a way  to make a difference between `default` import and not default import :
```
import something from "somewhere"; // this actually imports whatever the *default* import is as `something`
import * as something from "somewhere"; // this imports all the exported things from the module as fields on the object `something` (in a hand wavy way)
```

We currently can't differentiate between those, and I don't know of this will be useful, but might be required if/when goja has import/export support

We can implement a thing even now though as the babel transformation [makes this possible](https://babeljs.io/docs/en/babel-plugin-transform-modules-commonjs#babel) 

WDYT @na-- @imiric @codebien